src/apisix/plugins/bk-header-rewrite.lua

```diff
@@ -0,0 +1,142 @@
+--
+-- TencentBlueKing is pleased to support the open source community by making
+-- 蓝鲸智云 - API 网关(BlueKing - APIGateway) available.
+-- Copyright (C) 2017 THL A29 Limited, a Tencent company. All rights reserved.
+-- Licensed under the MIT License (the "License"); you may not use this file except
+-- in compliance with the License. You may obtain a copy of the License at
+--
+--     http://opensource.org/licenses/MIT
+--
+-- Unless required by applicable law or agreed to in writing, software distributed under
+-- the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+-- either express or implied. See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+-- We undertake not to change the open source license (MIT license) applicable
+-- to the current version of the project delivered to anyone in the future.
+--
+
+-- bk-header-rewrite
+--
+-- Rewrite the headers of a request using the plugin configuration.
+
+local core        = require("apisix.core")
+local plugin_name = "bk-header-rewrite"
+local pairs       = pairs
+
+local lrucache = core.lrucache.new({
+    type = "plugin",
+})
+
+local schema = {
+    type = "object",
+    description = "new headers for request",
+    minProperties = 1,
+    additionalProperties = false,
+    properties = {
+        add = {
+            type = "object",
+            minProperties = 1,
+            patternProperties = {
+                ["^[^:]+$"] = {
+                    oneOf = {
+                        { type = "string" },
+                        { type = "number" }
+                    }
+                }
+            },
+        },
+        set = {
+            type = "object",
+            minProperties = 1,
+            patternProperties = {
+                ["^[^:]+$"] = {
+                    oneOf = {
+                        { type = "string" },
+                        { type = "number" },
+                    }
+                }
+            },
+        },
+        remove = {
+            type = "array",
+            minItems = 1,
+            items = {
+                type = "string",
+                -- "Referer"
+                pattern = "^[^:]+$"
+            }
+        },
+    },
+}
+
+
+local _M = {
+    version  = 0.1,
+    priority = 17420,
+    name     = plugin_name,
+    schema   = schema,
+}
+
+function _M.check_schema(conf)
+    local ok, err = core.schema.check(schema, conf)
+    if not ok then
+        return false, err
+    end
+
+    return true
+end
+
+
+local function create_header_operation(hdr_conf)
```

Comment:
- add comment for this function
- rename `hdr_conf` to `header_conf` (more clear)